File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/repository/PostsRepository.java
package com.example.postscommentsapp.domain.repository;

import com.example.postscommentsapp.domain.model.Post;

import java.util.List;

public interface PostsRepository {

    List<Post> getPosts() throws Exception;

    List<Post> searchPosts(String query) throws Exception;
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/domain/repository/PostsRepository.kt
package com.example.postscommentsapp.domain.repository

import com.example.postscommentsapp.domain.model.Post

interface PostsRepository {
    suspend fun getPosts(): List<Post>
    suspend fun getPostById(postId: Int): Post?
    suspend fun searchPosts(query: String): List<Post>
}
